packages/assets-controllers/src/AccountTrackerController.ts

```diff
@@ -150,20 +162,50 @@ export class AccountTrackerController extends BaseController<
   }
 
   /**
-   * Refreshes all accounts in the current keychain.
+   * Refreshes the balances of the accounts depending on the multi-account setting.
+   * If multi-account is disabled, only updates the selected account balance.
+   * If multi-account is enabled, updates balances for all accounts.
+   *
+   * @async
    */
   refresh = async () => {
     this.syncAccounts();
     const accounts = { ...this.state.accounts };
+    const isMultiAccountBalancesEnabled = this.getMultiAccountBalancesEnabled();
+
+    if (!isMultiAccountBalancesEnabled) {
+      const selectedAddress = this.getSelectedAddress();
+      const balance = await this.getBalanceFromChain(selectedAddress);
+      accounts[selectedAddress] = { balance: BNToHex(balance) };
+      this.update({ accounts });
+      return;
+    }
+
     for (const address in accounts) {
-      await safelyExecuteWithTimeout(async () => {
-        const balance = await query(this.ethQuery, 'getBalance', [address]);
-        accounts[address] = { balance: BNToHex(balance) };
-      });
+      const balance = await this.getBalanceFromChain(address);
+      accounts[address] = { balance: BNToHex(balance) };
     }
+
     this.update({ accounts });
   };
 
+  /**
+   * Fetches the balance of a given address from the blockchain.
+   *
+   * @async
+   * @param address - The account address to fetch the balance for.
+   * @returns A promise that resolves to the balance in a hex string format.
+   */
+  async getBalanceFromChain(address: string): Promise<string | undefined> {
```

Comment:
Nit: Have you considered making this a private method?

There is no problem with this being a public method, if there is a use for it. It's just nice to keep things private whenever possible (less API surface area to test and maintain).